openedx/core/djangoapps/team_api/tests/test_views.py

```diff
@@ -0,0 +1,437 @@
+        user = user if user else self.student_user
+        self.client.login(username=user.username, password=self.test_password)
+        response = self.client.get(reverse('teams_detail', args=[team_id]))
+        self.assertEquals(response.status_code, expected_status)
+        return response
+
+    def get_team_detail_json(self, team_id, expected_status, user=None):
+        response = self.get_team_detail(team_id, expected_status, user=user)
+        return json.loads(response.content)
+
+    def test_detail_team_anonymous(self):
+        response = self.client.get(reverse('teams_detail', args=[self.test_team_1.team_id]))
+        self.assertEquals(response.status_code, 403)
+
+    def test_detail_team_logged_in_not_active(self):
+        self.setup_inactive_user()
+        response = self.client.get(reverse('teams_detail', args=[self.test_team_1.team_id]))
+        self.assertEquals(response.status_code, 200)
+
+    def test_detail_team_logged_in(self):
+        team = self.get_team_detail_json(self.test_team_1.team_id, 200)
+        self.assertEquals(team['description'], self.test_team_1.description)
+
+    def test_detail_team_does_not_exist(self):
+        self.get_team_detail('foobar', 404)
+
+    def patch_team_detail(self, team_id, expected_status, user=None, data=None):
+        user = user if user else self.student_user
+        data = data if data else {}
+        data = json.dumps(data)
+        self.client.login(username=user.username, password=self.test_password)
+        response = self.client.patch(
+            reverse('teams_detail', args=[team_id]),
+            data=data,
+            content_type='application/merge-patch+json',
+        )
+        self.assertEquals(response.status_code, expected_status)
+
+    def test_update_team_anonymous(self):
+        response = self.client.patch(reverse('teams_detail', args=[self.test_team_1.team_id]))
+        self.assertEquals(response.status_code, 403)
+
+    def test_update_team_logged_in(self):
+        self.patch_team_detail(self.test_team_1.team_id, 403)
+
+    def test_update_team_staff(self):
+        self.patch_team_detail(self.test_team_1.team_id, 200, user=self.staff_user, data={'name': 'foo'})
+        team = self.get_team_detail_json(self.test_team_1.team_id, 200, user=self.staff_user)
+        self.assertEquals(team['name'], 'foo')
+
+    def test_update_team_extra_fields(self):
+        self.patch_team_detail(self.test_team_1.team_id, 400, user=self.staff_user, data={'foo': 'bar'})
+
+    def test_update_team_read_only_fields(self):
+        self.patch_team_detail(self.test_team_1.team_id, 400, user=self.staff_user, data={'team_id': 'foobar'})
+
+    def test_update_team_blank_description(self):
+        self.patch_team_detail(self.test_team_1.team_id, 400, user=self.staff_user, data={'description': ''})
+
+    def test_update_team_does_not_exist(self):
+        self.patch_team_detail('foobar', 404, user=self.staff_user)
+
+
+class TestTopicsAPI(TestTeamAPI):
```

Comment:
Yeah, I had originally thought that it might make sense to break things out a bit but now I'm realizing we should either (a) just move all tests into one class, or (b) split into separate classes where only non-tests exist in the superclass. I suspect option (a) is the right one, since `TestTopicsAPI` only really shares the `setUp` code.